### PR TITLE
Update `MessageTemplates` generation to work with Smarty3

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Update `MessageTemplates` generation to not use weird old methods which don't work with Smarty3 & make the code really hard to read

Before
----------------------------------------
The message template generation code assigns various information to the template & the template calls smarty->fetch() internally to get the contents

After
----------------------------------------
The template generation code assigns the contents to the tpl

I did an after-regen and the only changes relevant to affected tables are in the event template - which has un-regened changes - which I will put up separately here https://github.com/civicrm/civicrm-core/pull/28804

image

Technical Details
----------------------------------------
the `$smarty` loop stuff is hard to parse but it was the smarty->fetch() stuff that was crashing when smarty3 was used for `genCode`

Once I was about 2/3 of the way through I remembered all the reasons why it would be better to switch this stuff to using managed entities - but I recall when I was going to do that before Tim said I couldn't use an mgd file & then Seamus did most of the work on these files so I didn't attempt it - but it would give us revert for free....

Comments
----------------------------------------

### CODE REVIEW

**Comment by eileenmcnaughton** [comment]
The regen shows this is *almost* the same now - 2 areas of change

1) event_online receipt - this has changes since the last regen so the fact only this changed of the message templates is a good sign

2) the sample templates went missing  - investigating
image

**Comment by colemanw** [comment]
Ok tests are happy with this.
